Dockerfile

```diff
@@ -559,7 +564,8 @@ RUN --mount=type=cache,sharing=locked,id=moby-dev-aptlib,target=/var/lib/apt \
             libseccomp-dev \
             libsecret-1-dev \
             libsystemd-dev \
-            libudev-dev
+            libudev-dev \
+            yamllint=${YAMLLINT_VERSION}\*
```

Comment:
This is going to cause problems over time -- the Debian archive isn't going to keep old versions, so instead of pinning to a specific version forever, this is going to ensure that we eventually just fail to build (although likely not until the next Debian release unless something _really_ exceptional happens, so I guess we're mostly probably OK for now?)

Reply:
Good point. I think we're pinning this one so that we didn't get random linter updates causing CI to fail, but we should definitely keep it a bit up to date periodically.

Still looking as well if there's a good alternative not involving python. I think we're mainly interested in well-formed-ness (need to check again where we're all using it)

Reply:
- opened https://github.com/moby/moby/pull/46884 for this